spring-cloud-kubernetes-integration-tests/spring-cloud-kubernetes-fabric8-client-reload/src/test/java/org/springframework/cloud/kubernetes/fabric8/configmap/event/reload/BootstrapEnabledPollingReloadConfigMapMountDelegate.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2013-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.kubernetes.fabric8.configmap.event.reload;
+
+import java.io.InputStream;
+import java.time.Duration;
+import java.util.Map;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.utils.Serialization;
+import org.junit.jupiter.api.Assertions;
+import org.testcontainers.k3s.K3sContainer;
+
+import org.springframework.cloud.kubernetes.integration.tests.commons.Commons;
+import org.springframework.cloud.kubernetes.integration.tests.commons.fabric8_client.Util;
+import org.springframework.http.HttpMethod;
+import org.springframework.web.reactive.function.client.WebClient;
+
+import static org.awaitility.Awaitility.await;
+import static org.springframework.cloud.kubernetes.fabric8.configmap.event.reload.TestUtil.builder;
+import static org.springframework.cloud.kubernetes.fabric8.configmap.event.reload.TestUtil.retrySpec;
+
+final class BootstrapEnabledPollingReloadConfigMapMountDelegate {
+
+	/**
+	 * <pre>
+	 *     - we have bootstrap enabled, which means we will 'locate' property sources
+	 *       from config maps.
+	 *     - there are no explicit config maps to search for, but what we will also read,
+	 *     	 is 'spring.cloud.kubernetes.config.paths', which we have set to
+	 *     	 '/tmp/application.properties'
+	 *       in this test. That is populated by the volumeMounts (see deployment-mount.yaml)
+	 *     - we first assert that we are actually reading the path based source via (1), (2) and (3).
+	 *
+	 *     - we then change the config map content, wait for k8s to pick it up and replace them
+	 *     - our polling will then detect that change, and trigger a reload.
+	 * </pre>
+	 */
+	static void testPollingReloadConfigMapWithBootstrap(KubernetesClient client, Util util, K3sContainer container,
+			String appLabelValue) {
+		// (1)
+		Commons.waitForLogStatement("paths property sources : [/tmp/application.properties]", container, appLabelValue);
+		// (2)
+		Commons.waitForLogStatement("will add file-based property source : /tmp/application.properties", container,
+				appLabelValue);
+		// (3)
+		WebClient webClient = builder().baseUrl("http://localhost/key").build();
```

Comment:
Why use WebClient instad of RestClient?

Reply:
Huh, I guess I never noticed.  It just seemed odd to me to use WebClient and then block when you could just use RestTemplate

Reply:
RestTemplate isn't going anywhere.  Anyways it is no big deal, its a test